src/main/java/io/aiven/kafka/connect/opensearch/BulkProcessor.java

```diff
@@ -594,4 +670,53 @@ public String toString() {
             return name().toLowerCase(Locale.ROOT);
         }
     }
+
+    public enum ErrorToleranceType {
+        NONE,
+        ALL;
+
+        public static final ErrorToleranceType DEFAULT = NONE;
+
+        // Want values for "errors.tolerance" property to be case-insensitive
+        public static final ConfigDef.Validator VALIDATOR = new ConfigDef.Validator() {
+            private final ConfigDef.ValidString validator = ConfigDef.ValidString.in(names());
+
+            @Override
+            public void ensureValid(final String name, final Object value) {
+                if (value instanceof String) {
+                    final String lowerCaseStringValue = ((String) value).toLowerCase(Locale.ROOT);
+                    validator.ensureValid(name, lowerCaseStringValue);
+                } else {
+                    validator.ensureValid(name, value);
+                }
+            }
+
+            // Overridden here so that ConfigDef.toEnrichedRst shows possible values correctly
+            @Override
+            public String toString() {
+                return validator.toString();
+            }
+
+        };
+
+        public static String[] names() {
+            final ErrorToleranceType[] errorToleranceTypes = values();
+            final String[] result = new String[errorToleranceTypes.length];
+
+            for (int i = 0; i < errorToleranceTypes.length; i++) {
+                result[i] = errorToleranceTypes[i].toString();
+            }
+
+            return result;
+        }
+
+        public static ErrorToleranceType forValue(final String value) {
+            return valueOf(value.toUpperCase(Locale.ROOT));
+        }
+
+        @Override
+        public String toString() {
+            return name().toLowerCase(Locale.ROOT);
+        }
+    }
```

Comment:
Similar to comments in the config class--we shouldn't be redefining this logic in the connector.
```suggestion
```